cpp/src/arrow/flight/sql/odbc/odbc_api.cc

```diff
@@ -61,18 +60,18 @@ SQLRETURN SQLAllocHandle(SQLSMALLINT type, SQLHANDLE parent, SQLHANDLE* result)
         return SQL_INVALID_HANDLE;
```

Comment:
Perhaps we can remove the checks for null handle. `ExecuteWithDiagnostics` already has a check for this